pkg/forwarder/backoff.go

```diff
@@ -9,25 +9,75 @@ import (
 	"math"
 	"math/rand"
 	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	log "github.com/cihub/seelog"
 )
 
-const (
+const secondsFloat = float64(time.Second)
+
+var (
 	// This controls the overlap between consecutive retry interval ranges. When
 	// set to `2`, there is a guarantee that there will be no overlap. The overlap
 	// will asymptotically approach 50% the higher the value is set.
-	minBackoffFactor = 2
+	minBackoffFactor float64
 
 	// This controls the rate of exponential growth. Also, you can calculate the start
 	// of the very first retry interval range by evaluating the following expression:
 	// baseBackoffTime / minBackoffFactor * 2
-	baseBackoffTime = 2
+	baseBackoffTime float64
 
 	// This is the maximum number of seconds to wait for a retry.
-	maxBackoffTime = 64
+	maxBackoffTime float64
 
-	secondsFloat = float64(time.Second)
+	// This controls how many retry interval ranges to step down for an endpoint
+	// upon success. Increasing this should only be considered when maxBackoffTime
+	// is particularly high or if our intake team is particularly confident.
+	recoveryInterval int
 )
 
+func init() {
```

Comment:
I think the tests should be updated to use the configuration and also test the min/max limit we impose here.